include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1123,25 +1156,33 @@ __parallel_partition_copy(oneapi::dpl::__internal::__device_backend_tag __backen
                           _Range1&& __rng, _Range2&& __result, _UnaryPredicate __pred)
 {
     oneapi::dpl::__internal::__difference_t<_Range1> __n = __rng.size();
+#if _ONEDPL_COMPILE_KERNEL && _ONEDPL_KERNEL_BUNDLE_PRESENT
     if (oneapi::dpl::__par_backend_hetero::__is_gpu_with_sg_32(__exec))
     {
         using _GenMask = oneapi::dpl::__par_backend_hetero::__gen_mask<_UnaryPredicate>;
         using _WriteOp =
             oneapi::dpl::__par_backend_hetero::__write_to_id_if_else<oneapi::dpl::__internal::__pstl_assign>;
-
-        return __parallel_reduce_then_scan_copy(__backend_tag, std::forward<_ExecutionPolicy>(__exec),
-                                                std::forward<_Range1>(__rng), std::forward<_Range2>(__result), __n,
-                                                _GenMask{__pred}, _WriteOp{}, /*_IsUniquePattern=*/std::false_type{});
+        // Compile the kernels to check if the sub-group size is 32. This should not be necessary per the SYCL spec
+        // but is needed to check for an IGC workaround for a hardware bug where kernels may be compiled with a
+        // sub-group size of 16 despite requiring 32. If the wrong sub-group size is used, then fallback to
+        // multi-pass scan.
+        __reduce_then_scan_copy_kernels<_ExecutionPolicy, _Range1, _Range2, _GenMask,
+                                        oneapi::dpl::__internal::__difference_t<_Range1>, _WriteOp, std::false_type>
```

Comment:
similar to above (comment or pull up with using) to avoid the unlabeled `false_type`